site/components/src/main/java/uk/nhs/digital/website/beans/Video.java

```diff
@@ -16,7 +18,50 @@ public HippoHtml getIntroduction() {
 
     @HippoEssentialsGenerated(internalName = "website:externallinkbase")
     public String getVideoUri() {
-        return getSingleProperty("website:externallinkbase");
+        String videoUri = getSingleProperty("website:externallinkbase");
+
+        // Regex adapted from https://regexr.com/3dj5t - named groups added
+        Pattern youtubeTest = Pattern.compile(
+            "^(?<protocol>(?:https?:)?\\/\\/)"
+                + "?(?<subdomain>(?:www|m)\\.)"
+                + "?(?<domain>(?:youtube\\.com|youtu.be))"
+                + "(?<type>\\/(?:[\\w\\-]+\\?v=|embed\\/|v\\/)?)"
+                + "(?<videoID>[\\w\\-]+)"
+                + "(?<query>\\S+)?$",
+            Pattern.CASE_INSENSITIVE | Pattern.MULTILINE
+        );
+        Matcher youtubeMatch = youtubeTest.matcher(videoUri);
+
+        // Make sure that YouTube URLs use the right format
+        if (youtubeMatch.matches()) {
+            return String.format(
+                "https://www.youtube-nocookie.com/embed/%s?modestbranding=1",
+                youtubeMatch.group("videoID")
+            );
+        }
+
+        // Regex adapted from https://regexr.com/393r3 - named groups added
+        Pattern vimeoTest = Pattern.compile(
+            "(?:<iframe [^>]*src=\")?"
+                + "(?:https?:\\/\\/"
+                + "(?:[\\w]+\\.)*vimeo\\.com"
+                + "(?:[\\/\\w:]*(?:\\/videos)?)?\\/"
+                + "(?<videoID>[0-9]+)[^\\s]*)\"?"
+                + "(?:[^>]*><\\/iframe>)?"
+                + "(?:<p>.*<\\/p>)?",
+            Pattern.CASE_INSENSITIVE | Pattern.MULTILINE
+        );
+        Matcher vimeoMatch = vimeoTest.matcher(videoUri);
+
+        // Make sure that Vimeo URLs use the right embed format
+        if (vimeoMatch.matches()) {
+            return String.format(
+                "https://player.vimeo.com/video/%s?color=005eb8&portrait=0",
+                vimeoMatch.group("videoID")
+            );
+        }
```

Comment:
Same as above for YouTube. Should probably extract this Vimeo one  into a static helper. But it's okay for now.